var/spack/repos/builtin/packages/tensorflow/package.py

```diff
@@ -0,0 +1,377 @@
+
+    depends_on('cuda', when='+cuda')
+    depends_on('cudnn', when='+cuda')
+    depends_on('nccl', when='+nccl')
+
+    patch('url-zlib.patch',  when='@0.10.0')
+    patch('crosstool.patch', when='@0.10.0+cuda')
+    # Avoid build error: "no such package '@io_bazel_rules_docker..."
+    patch('io_bazel_rules_docker2.patch', when='@1.15.0,2.0.0')
+    # Avoide build error: "name 'new_http_archive' is not defined"
+    patch('http_archive.patch', when='@1.12.3')
+
+    phases = ['configure', 'install']
+
+    # https://www.tensorflow.org/install/source
+    def setup_build_environment(self, env):
+        spec = self.spec
+
+        if '+gcp' in spec:
+            env.set('TF_NEED_GCP', '1')
+        else:
+            env.set('TF_NEED_GCP', '0')
+
+        env.set('PYTHON_BIN_PATH', spec['python'].command.path)
+        env.set('SWIG_PATH', spec['swig'].prefix.bin)
+        env.set('GCC_HOST_COMPILER_PATH', spack_cc)
+
+        # CUDA related config options - note: tf has only been tested for cpu
+        if '+cuda' in spec:
+            env.set('GCC_HOST_COMPILER_PATH', self.compiler.cc)
+            env.set('TF_NEED_CUDA', '1')
+            env.set('TF_CUDA_VERSION', spec['cuda'].version.up_to(2))
+            env.set('TF_CUDNN_VERSION', spec['cudnn'].version.up_to(1))
+            # TODO also consider the case of cuda enabled but nccl disabled
+            env.set('TF_NCCL_VERSION', spec['nccl'].version.up_to(1))
+            if self.spec.satisfies('@1.14.0:'):
+                env.set('TF_CUDA_PATHS', ','.join([
+                    spec['cuda'].prefix,
+                    spec['nccl'].prefix,
+                    spec['cudnn'].prefix,
+                ]))
+            env.set('CUDA_TOOLKIT_PATH', spec['cuda'].prefix)
+            # ignored? as of tf@1.14.0:
+            env.set('CUDNN_INSTALL_PATH', spec['cudnn'].prefix)
+            # TODO: create a string valued variant for compute capabilities?
+            # one should be able to specify single or multiple capabilities
+            env.set('TF_CUDA_COMPUTE_CAPABILITIES', "6.1,7.5")
+
+            # @v1.13, config hangs without the following nccl env variables
+            # however, in the end it ignores them, and sets these incorrectly
+            # Because of this, these paths are reset via file filtering
+            # As shown in the "post_config_fix" section
+            env.set('NCCL_INSTALL_PATH', spec['nccl'].prefix)
+            env.set('NCCL_HDR_PATH', spec['nccl'].prefix.include)
+
+            env.set('TF_CUDA_CLANG', '0')
+            env.set('TF_NEED_ROCM', '0')
+            env.set('TF_NEED_TENSORRT', '0')
+        else:
+            env.set('TF_NEED_CUDA', '0')
+            env.set('TF_CUDA_VERSION', '')
+            env.set('CUDA_TOOLKIT_PATH', '')
+            env.set('TF_CUDNN_VERSION', '')
+            env.set('CUDNN_INSTALL_PATH', '')
+
+        # config options for version 1.0
+        if self.spec.satisfies('@1.0.0:'):
```

Comment:
If I set any of these env vars when building older versions of TF, will it complain? If not, maybe we could just unconditionally set them without checking the version number.

Reply:
I don't think TF will complain about it, rather it will disregard those options. So, probably there is no need for `if` statements there.